sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/joins.scala

```diff
@@ -144,6 +144,8 @@ object EliminateOuterJoin extends Rule[LogicalPlan] with PredicateHelper {
     val emptyRow = new GenericInternalRow(attributes.length)
     val boundE = BindReferences.bindReference(e, attributes)
     if (boundE.exists(_.isInstanceOf[Unevaluable])) return false
+    // don't evaluate an expression that might purposefully throw an exception
+    if (boundE.exists(_.isInstanceOf[RaiseError])) return false
     val v = boundE.eval(emptyRow)
```

Comment:
can we do a try-catch and swallow the error here? `RaiseError` doesn't match all the expressions that may fail during evaluation.

Reply:
>can we do a try-catch and swallow the error here?

Looking...

Reply:
>RaiseError doesn't match all the expressions that may fail during evaluation.

I see what you mean. This query also fails in `EliminateOuterJoin` (because you can't have a null key):
```
select *
from (select id, id as b from range(0, 10)) l
left outer join (select id, id + 1 as c from range(0, 10)) r
on l.id = r.id
where map(c, '2') is not null;
```
I will add a try/catch.